tools/cldr-apps/src/main/java/org/unicode/cldr/web/SubtypeToURLMap.java

```diff
@@ -0,0 +1,396 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package org.unicode.cldr.web;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.io.Reader;
+import java.io.StringReader;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.unicode.cldr.test.CheckCLDR.CheckStatus.Subtype;
+import org.unicode.cldr.util.CLDRConfig;
+import org.unicode.cldr.util.CLDRTool;
+
+import com.ibm.icu.text.MessageFormat;
+
+@CLDRTool(alias="subtype-to-url-map", description = "parse each of the params as a path or URL to a subtype map and check.")
+public class SubtypeToURLMap {
+    /**
+     * Little tool for validating input data.
+     * @param args list of files to validate, if empty runs against default data.
+     * @throws IOException
+     * @throws FileNotFoundException
+     */
+     public static void main(String args[]) throws FileNotFoundException, IOException {
+        if(args.length == 0) {
+            System.err.println("Usage: SubtypeToURLMap (url or file path). The default map is " + DEFAULT_URL);
+            return;
+        } else {
+            int problems = 0;
+            for(final String fn : args) {
+                System.out.println("data: " + fn);
+                SubtypeToURLMap map = getInstance(new File(fn));
+                problems += map.dump();
+            }
+            if(problems > 0) {
+                throw new IllegalArgumentException(MessageFormat.format("Total problem(s) found: {0} in {1} items(s)", problems, args.length));
+            }
+        }
+    }
+
+    /**
+     * Map from Subtype to URL
+     */
+    private Map<Subtype, String> map;
+
+    /**
+     * The set of URLs is kept as a List so the original order is retained.
+     */
+    private List<String> urlList;
+
+    /**
+     * Internal constructor for
+     * @param utf8Data for cleaned input data (i.e. not HTML)
+     * @throws IllegalArgumentException
+     */
+    protected SubtypeToURLMap(BufferedReader utf8Data) throws IllegalArgumentException{
```

Comment:
this constructor is too long to see all at once on the screen, making the logic harder to check; it could be shortened with subroutines, maybe an object containing the variables newMap ... urlCount

just a suggestion for future improvement, not a blocker for this review